src/microsim/schema/simulation.py

```diff
@@ -93,6 +93,28 @@ def _resolve_spaces(self) -> "Self":
             self.output_space.reference = self.truth_space
         return self
 
+    @model_validator(mode="after")
+    def _check_fluorophores_equal_in_samples(self) -> "Self":
+        fp_names = [{lbl.fluorophore.name for lbl in s.labels} for s in self.samples]
+        if len({frozenset(s) for s in fp_names}) != 1:
+            raise ValueError(
+                "All samples in the batch must use the same set of fluorophores."
+            )
+        return self
+
+    @field_validator("samples")
+    def _samples_to_list(value: Sample | list[Sample]) -> list[Sample]:
+        return [value] if isinstance(value, Sample) else value
```

Comment:
```suggestion
    @field_validator("samples", mode="before")
    @classmethod
    def _samples_to_list(cls, value: Any) -> Sequence[Any]:
        return [value] if not isinstance(value, (list, tuple)) else value
```

1. makes it a classmethod for clarity (pydantic would do this anyway)
2. adds the cls, argument to the signature
3. because it's a "before" validator, all we really need to do is make sure we're passing a list to pydantic's "actual" validation, which will cast everything to a Sample